system/services/systemAlerts/SystemAlertsService.cfc

```diff
@@ -0,0 +1,459 @@
+/**
+ * @presideService true
+ * @singleton      true
+ */
+component {
+
+// CONSTRUCTOR
+	/**
+	 * @validAlertLevels.inject  coldbox:setting:enum.systemAlertLevel
+	 *
+	 */
+	public any function init( required array validAlertLevels ) {
+		_setValidAlertLevels( arguments.validAlertLevels );
+
+		return this;
+	}
+
+	public void function setupSystemAlerts() {
+		_discoverSystemAlertTypes();
+	}
+
+// PUBLIC API METHODS
+	public any function runCheck( required string type, string reference="", boolean async=true ) {
+		var checkHandler = "admin.systemAlerts.#arguments.type#.runCheck";
+		var config       = getAlertConfig( arguments.type );
+
+		if ( !$getColdbox().handlerExists( checkHandler ) ) {
+			return;
+		}
+
+		if ( arguments.async ) {
+			$createTask(
+				  event             = "admin.systemAlerts.runCheckInBackgroundThread"
+				, args              = { type=arguments.type, reference=arguments.reference }
+				, runNow            = true
+				, discardOnComplete = false
+			);
+			return;
+		}
+
+		var checks     = [];
+		var references = config.isMultiCheck && !Len( arguments.reference ) ? $runEvent(
```

Comment:
This feels a little loose. i.e. if the `references` handler action does not exist, the error won't be all that instructional for a developer.

Should we be checking that the handler exists first and then throwing some explicit error when expecting references and we don't have them (either because not passed or no handler exists, or handler not returning correct/any data?